storage_service/common/management/commands/import_aip.py

```diff
@@ -43,9 +43,9 @@
     -rwxrwxr-x  1 joeldunham  wheel    51M 30 Jul 22:58 /tmp/am-pipeline-data/www/AIPsStore/237e/12b4/03c9/451a/9be7/4915/b0bd/9012/FakeCVA2-237e12b4-03c9-451a-9be7-4915b0bd9012.7z
 
 """
-import glob
 import logging
 import os
+import pathlib
```

Comment:
I think the logic of this command (and probably others) is very well abstracted from the rest of the application which gives us more freedom.

The change you propose here of converting several strings into paths (granted, it's the approach we've been following so far) might be unnecessary if we start with paths from the beginning. Specifically from the point where the command parses its arguments:

```python
        parser.add_argument("aip_path", help="Full path to the AIP to be imported", type=pathlib.Path)
```

That would give you a path from the beginning which you only convert to strings when it's necessary.

It reminds me of [Ned Batchelder's "Unicode Sandwich"](https://nedbatchelder.com/text/unipain.html). The concept of handling native types inside the application which you decode as soon as possible and encode as late as possible was really useful for porting strings from Python 2 to 3 and more recently when we migrated strings to UUIDs.

Reply:
Thank you for the insightful suggestion. I appreciate the parallel drawn to the "Unicode Sandwich" concept and agree with it. I've implement this approach to minimize unnecessary string conversions. Please have another look when you get a chance.

In [this part](https://github.com/artefactual/archivematica-storage-service/blob/7559a8ee151736e082a4901a354e3c32cd9b30c2/storage_service/common/management/commands/import_aip.py#L388-L393) I have more doubts due to its link with models/package, models/location and utils. If I focus on them now or on the next PRs.

Reply:
This looks good. We can address cleanups in future PRs :+1: